spec/javascripts/stimulus/popout_controller_test.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { Application } from "stimulus";
+import popout_controller from "../../../app/webpacker/controllers/popout_controller";
+
+describe("PopoutController", () => {
+  beforeAll(() => {
+    const application = Application.start();
+    application.register("popout", popout_controller);
+  });
+
+  beforeEach(() => {
+    document.body.innerHTML = `
+      <div data-controller="popout">
+        <button id="button" data-popout-target="button">On demand</button>
+        <div id="dialog" data-popout-target="dialog" style="display: none;">
+          <input id="input1">
+          <input id="input2">
+        </div>
+      </div>
+    `;
+
+    const button = document.getElementById("button");
+    const input1 = document.getElementById("input1");
+    const input2 = document.getElementById("input2");
+  });
+
+  describe("Show", () => {
+    it("shows the dialog on click", () => {
+      button.click();
+
+      expect(dialog.style.display).toBe("block"); // visible
```

Comment:
I can't figure out where `dialog` is coming from ? I ran the test and it's passing, and I can see it refers to the `<div>` with id `dialog` but I would expect so see someting like this somewhere : 
```javascript
const dialog = document.getElementById("dialog");
```

Reply:
Oh, it turns out the variables defined in `beforeEach` aren't used at all, and Jest is automatically creating them based on the ids in the document! I can't find reference to this in Jest documentation.

It's funny, I was thinking yesterday that it would be really handy to create a helper to do that, but didn't have time. I'm glad I didn't bother.